WcaOnRails/app/controllers/database_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class DatabaseController < ApplicationController
+  EXPORT_PUBLIC_FOLDER = Rails.root.join('public', 'export')
+
+  RESULTS_EXPORT_FOLDER = EXPORT_PUBLIC_FOLDER.join('results')
+  DEVELOPER_EXPORT_FOLDER = EXPORT_PUBLIC_FOLDER.join('developer')
+
+  RESULTS_SQL = "WCA_export.sql"
+  RESULTS_README = "README.md"
+  RESULTS_METADATA = "metadata.json"
+
+  RESULTS_SQL_PERMALINK = "#{RESULTS_SQL}.zip".freeze
```

Comment:
Just searched what `freeze` is, and it is used to ensure object cannot be modified. I'm bit curious why it's applicable for this file alone? 😅

Reply:
It is applicable for strings in any file based on the `# frozen_string_literal: true` line at the top of the file.
This string that you marked is an exception because it contains interpolation (basically another string inserted into the string with `#{}`" and the `frozen_string_literal` doesn't apply because then it's not a **literal** anymore.
So I have to explicitly freeze it while all other "straight-forward-strings" in the file are implicitly frozen.